### ISSUE TITLE
spectrum2 fails to build on FreeBSD

### ISSUE THREAD

**Issue by drook** [hanzz/spectrum2#109]
spectrum2 fails to build on FreeBSD

FreeBSD:

````FreeBSD
# uname -a
FreeBSD elf.hq.norma.perm.ru 10.3-PRERELEASE FreeBSD 10.3-PRERELEASE #0 r294405: Thu Jan 21 14:03:07 YEKT 2016     emz@elf.hq.norma.perm.ru:/usr/obj/usr/src/sys/ELF  amd64
````

Compiler:
````clang
# clang -v
FreeBSD clang version 3.4.1 (tags/RELEASE_34/dot1-final 208032) 20140512
Target: x86_64-unknown-freebsd10.3
Thread model: posix
Selected GCC installation:
````

Output:
````build.log
[ 96%] Built target spectrum2_skype_backend
[ 97%] Linking CXX executable spectrum2_manager
CMakeFiles/spectrum2_manager.dir/server.cpp.o: In function `Server::Server(ManagerConfig*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)':
/home/emz/spectrum/spectrum2-2.0.1/spectrum_manager/src/server.cpp:(.text+0x1092): undefined reference to `mg_set_ssl'
c++: error: linker command failed with exit code 1 (use -v to see invocation)
spectrum_manager/src/CMakeFiles/spectrum2_manager.dir/build.make:241: recipe for target 'spectrum_manager/src/spectrum2_manager' failed
gmake[2]: *** [spectrum_manager/src/spectrum2_manager] Error 1
CMakeFiles/Makefile2:1158: recipe for target 'spectrum_manager/src/CMakeFiles/spectrum2_manager.dir/all' failed
gmake[1]: *** [spectrum_manager/src/CMakeFiles/spectrum2_manager.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
gmake: *** [all] Error 2
````

Plus (a minor issue) would be nice if spectrum2 would include /usr/local/lib in it's linker path when building. Had to hack into the link.txt files.

**Comment by vitalyster** [comment]
> mg_set_ssl
That is reference from mongoose.cpp
> /usr/local/lib in it's linker path
how other CMake programs do it?

**Comment by drook** [comment]
Thanks, cap (you know, if I want some evident information, I could use `find`).
"Other programs" on FreeBSD do it in 100% of cases.

**Comment by hanzz** [comment]
Does it compile mongoose.cpp during the build for you?

Can you run the build with "VERBOSE=1 make" and send the output here?

**Comment by drook** [comment]
`mongoose.c` is building just fine, but `mongoose.c.o` doesn't have `mg_set_ssl` symbol exported for some reason. I asked for help big guys from the #freebsd channel, they told me that it's because `MG_ENABLE_SSL` isn't defined. They also told me to either define it with `-DMG_ENABLE_SSL` in the compiler invocation command, or in the `mongoose.h` header, which I did. This way spectrum2 was built successfully. Don't know if it's valid, at least manager is showing help. Right at this moment I'm fighting `log4cxx`, which complains with cryptic messages. Could you please explain why I have this problem with `MG_ENABLE_SSL` ? I'm not much of a programmer. Thanks.

**Comment by drook** [comment]
Without my modifications the building process looks as follows:

````building
[ 95%] Building C object spectrum_manager/src/CMakeFiles/spectrum2_manager.dir/mongoose.c.o
cd /home/emz/spectrum/spectrum2-2.0.1/spectrum_manager/src && /usr/bin/cc  -DBOOST_FILESYSTEM_VERSION=3 -DBOOST_SIGNALS_NO_DEPRECATION_WARNING -DPURPLE_RUNTIME=0 -DQT_NO_DEBUG -DSPECTRUM_VERSION=\"2.0.1\" -DSUPPORT_LEGACY_CAPS -DWITH_LIBEVENT -DWITH_LOG4CXX -DWITH_MYSQL -DWITH_PROTOBUF -DWITH_SQLITE -I/usr/local/include -I/usr/local/include/mysql -I/usr/local/include/libpurple -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include/dbus-1.0 -I/usr/local/lib/dbus-1.0/include -I/home/emz/spectrum/spectrum2-2.0.1/include -I/usr/local/swiften/include   -o CMakeFiles/spectrum2_manager.dir/mongoose.c.o   -c /home/emz/spectrum/spectrum2-2.0.1/spectrum_manager/src/mongoose.c
[ 97%] Linking CXX executable spectrum2_manager
cd /home/emz/spectrum/spectrum2-2.0.1/spectrum_manager/src && /usr/local/bin/cmake -E cmake_link_script CMakeFiles/spectrum2_manager.dir/link.txt --verbose=1
/usr/bin/CC CMakeFiles/spectrum2_manager.dir/managerconfig.cpp.o CMakeFiles/spectrum2_manager.dir/methods.cpp.o CMakeFiles/spectrum2_manager.dir/mongoose.c.o CMakeFiles/spectrum2_manager.dir/server.cpp.o CMakeFiles/spectrum2_manager.dir/main.cpp.o CMakeFiles/spectrum2_manager.dir/APIServer.cpp.o -o spectrum2_manager  ../../libtransport/libtransport.so.2.0 -L/usr/local/swiften/lib -L/usr/local/lib -lSwiften -lboost_signals -lboost_system -lboost_thread -lboost_regex -lboost_program_options -lboost_filesystem -lboost_serialization -lboost_date_time -lidn -lz -lssl -lcrypto -lexpat -lavahi-client -lavahi-common -lpthread -lm -lc -lstdc++ /usr/local/lib/libprotobuf.so -lpthread ../../plugin/cpp/libtransport-plugin.so.2.0 /usr/local/lib/libboost_program_options.so /usr/local/lib/libboost_date_time.so /usr/local/lib/libboost_system.so /usr/local/lib/libboost_filesystem.so /usr/local/lib/libboost_regex.so /usr/local/lib/libboost_thread.so /usr/local/lib/libboost_signals.so /usr/local/lib/libboost_locale.so /usr/local/lib/libcurl.so /usr/local/lib/libsqlite3.so /usr/local/lib/mysql/libmysqlclient.so -lSwiften -lboost_signals -lboost_system -lboost_thread -lboost_regex -lboost_program_options -lboost_filesystem -lboost_serialization -lboost_date_time -lidn -lz -lssl -lcrypto -lexpat -lavahi-client -lavahi-common -lpthread -lm -lc -lstdc++ /usr/local/lib/liblog4cxx.so /usr/local/lib/libpopt.so /usr/local/lib/libprotobuf.so -Wl,-rpath,/home/emz/spectrum/spectrum2-2.0.1/libtransport:/usr/local/lib:/home/emz/spectrum/spectrum2-2.0.1/plugin/cpp:/usr/local/lib/mysql: 
CMakeFiles/spectrum2_manager.dir/server.cpp.o: In function `Server::Server(ManagerConfig*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)':
/home/emz/spectrum/spectrum2-2.0.1/spectrum_manager/src/server.cpp:(.text+0x1092): undefined reference to `mg_set_ssl'
c++: error: linker command failed with exit code 1 (use -v to see invocation)
spectrum_manager/src/CMakeFiles/spectrum2_manager.dir/build.make:241: recipe for target 'spectrum_manager/src/spectrum2_manager' failed
gmake[2]: *** [spectrum_manager/src/spectrum2_manager] Error 1
gmake[2]: Leaving directory '/usr/home/emz/spectrum/spectrum2-2.0.1'
CMakeFiles/Makefile2:1158: recipe for target 'spectrum_manager/src/CMakeFiles/spectrum2_manager.dir/all' failed
gmake[1]: *** [spectrum_manager/src/CMakeFiles/spectrum2_manager.dir/all] Error 2
gmake[1]: Leaving directory '/usr/home/emz/spectrum/spectrum2-2.0.1'
Makefile:127: recipe for target 'all' failed
gmake: *** [all] Error 2
````

**Comment by vitalyster** [comment]
fixed in f747ef0b85f9
